handlers.go

```diff
@@ -15,10 +36,132 @@ func formatIndexPage(appURL string) string {
 	return page
 }
 
-// getIndexHandler returns an index handler that informs the visitor that this
-// host runs inside an enclave.
-func getIndexHandler(cfg *Config) http.HandlerFunc {
+// rootHandler returns a handler that informs the visitor that this host runs
+// inside an enclave.  This is useful for testing.
+func rootHandler(cfg *Config) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		fmt.Fprintln(w, formatIndexPage(cfg.AppURL))
 	}
 }
+
+// reqSyncHandler returns a handler that lets the enclave application request
+// state synchronization, which copies the given remote enclave's state into
+// our state.
+func reqSyncHandler(e *Enclave) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		q := r.URL.Query()
+		// The 'addr' parameter must have the following form:
+		// https://example.com:8443
+		addrs, ok := q["addr"]
```

Comment:
So the idea here is that the enclave application knows (somehow!) a list of other instances to contact for state transfer and it passes that in here?

I don't really like this; even if the peer list isn't something nitriding manages (and I think eventually it should be) it would be better to configure an allow list to validate against. This is a trusted API because it's on the enclave-private server, but it would be bad if it accidentally got routed to a public interface.

All of this is still a draft really, so maybe just mark it `FIXME` or open a follow-up issue for now?

Reply:
> So the idea here is that the enclave application knows (somehow!) a list of other instances to contact for state transfer and it passes that in here?

Yes. We haven't narrowed down the details yet (it depends on what's reasonable in the context of our k8s setup) but eV suggested to use k8s's built-in DNS SRV discovery: https://github.com/brave/nitriding/issues/44

I added a FIXME in 7b6f27e.